Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1804 @@
+
+Also note: 'imagemagick' in the default Ubuntu 16 repository does not supprt JPEG2000. Someone has made a patched repo (ppa:lyrasis/imagemagick-jp2) but it is now out of date. So if you need this functionality you may need to compile from source or force install an older version. The needed functionality is available from Openjpeg.
+
+```
+apt-get -y install oracle-java8-installer oracle-java8-set-default libjpeg-dev libpng12-dev libtiff5-dev php7.1 php7.1-cli php7.1-curl php7.1-dev php7.1-gd php7.1-ldap php7.1-mysql php7.1-mbstring php7.1-xsl php-apcu php-soap php-xml-htmlsax3 php-xml-parser php-xmlrpc php-xml-rpc2 php-xml-serializer php-imagick php7.1-mcrypt php-xml* mysql-server libmysql-java vim curl apache2 rsync wget imagemagick ant libimage-exiftool-perl unzip lame autoconf build-essential checkinstall git libass-dev libfaac-dev libgpac-dev libmp3lame-dev libopencore-amrnb-dev libopencore-amrwb-dev librtmp-dev libtheora-dev libtool libvorbis-dev pkg-config texi2html zlib1g-dev ffmpeg2theora poppler-utils python-pip libreoffice libreoffice-writer libreoffice-calc libreoffice-impress libreoffice-draw bibutils ufraw links monit tesseract-ocr tesseract-ocr-eng tesseract-ocr-fra tesseract-ocr-spa tesseract-ocr-ita tesseract-ocr-por tesseract-ocr-hin tesseract-ocr-deu tesseract-ocr-jpn tesseract-ocr-rus tomcat7 tomcat7-admin yasm jetty9 libx264-dev libfdk-aac-dev libopus-dev openjpeg-tools libopenjp2-tools drush unoconv rsnapshot maven apg ssl-cert fail2ban mailutils
+```
+
+You will be prompted three times for interactive installation, once from MySQL (provide a 'root' password), once from Postfix (choose 'no configuration') and once from Oracle Java (say yes to the license). Be sure to update your `islandora-install.properties` file with the MySQL password (DB_ROOT_PASSWORD). There will be an error from installing Tomcat7 (it won't start correctly). Ignore it at this time.
+
+You should run `update-alternatives --config php` and make sure it is set to `php7.1`. Else Drupal will not install. This is because we are using the 7.1 Apache module and 7.2 material may well have been installed and set to the default for the OS during the `apt-get install`.
+
+Run `mysql_secure_installation`. You will be prompted for the root password and answers to various questions. If you have created a strong root password at this time, feel free to enable the password validation plugin and set the strength requirements appropriately. You can always do this again later if you wish to use a weak password during the setup phase. Answer yes to all the other questions.
+
+### Software Dependencies Compiled from Source <a id="from-source"></a> 
+
+#### ghostscript  <a id="ghostscript"></a>
```

Comment:
If it works fine with the newer gs should this section be removed?

Reply:
I've verified that this works to the extent that the use of 'gs' to create pdf derivatives of
e.g. MS Word documents throws no errors and produces a readable pdf. But I haven't
tested it extensively (maybe 4 or 5 documents). Do you have a suite of tests I could
apply to verify more officially? Like some known problematic cases?

Reply:
Please see details here https://github.com/Islandora-Labs/islandora_vagrant/issues/127#issuecomment-330959453

Reply:
The test document isn't available any more (404).

Reply:
You should be able to replicate the issue using any PDF using the test cases discussed.

Reply:
I grabbed a pdf from the site, a newspaper page, and tried a few 'gs' converts on it,
including '-sDEVICE=tiff24nc', and it works fine. Not sure what the appropriate device
would be to test the issue.

Reply:
I guess I'm still puzzled what the input file is supposed to be. The output is definitely
a 'pdfa' archival pdf, not a tiff, as I assumed above. But if the input is a pdf with embedded
tiff, then this appears to work. But if the input is a tiff, then it fails.

Reply:
....and reading the original bug report more closely, I see I don't have quite the
right command yet. The problem is to produce a tiff OBJ file. I'll try to find the
command in the newspaper solution pack (looked there first but didn't find one).

Reply:
OK yet again. I think I've tracked down the command, in 'islandora_paged_content'.
The function is 'islandora_paged_content_extract_tiff_from_pdf', and the following
commands work with no error at all:

gs -q -dNOPAUSE -dBATCH -sDEVICE=tiff48nc -sCompression=none -sOutputFile=tst-300-48-p2.tiff -r300dpi -dFirstPage=1 -dLastPage=2 sample-multipage.pdf

gs -q -dNOPAUSE -dBATCH -sDEVICE=tiff48nc -sCompression=none -sOutputFile=tst-150-48-p2.tiff -r150dpi -dFirstPage=1 -dLastPage=2 sample-multipage.pdf

The multipage pdf is from the site mentioned in the bug report. And 'file' reports that the
results are TIFF images in RGB format.

The commands using 'tiff64nc' also work without error, and produce CMYK format output,
which look weird but display. ;)

Reply:
Ghostscript version is 9.18.

Reply:
The fact that this version of 'gs' causes a warning under some commands might
still prove an issue. I got no bubble up errors in my use of the pdf solution pack.
But maybe that needs more testing.

Reply:
So you confirmed that the derivatives still worked despite the warning using 48-bit RGB, 150 DPI or 300 DPI?

Reply:
Yes. There was no warning during this particular derivative creation (create a
tiff from a pdf). That came from the pdf solution pack, and involved the 'pdfwrite' device.

The error was caused by the use of this now deprecated flag: -dUseCIEColor.

Reply:
Further to the test: in addition to running the 'gs' command, I created a newspaper
through the web interface and added an issue and all derivatives were created correctly
(jp2,tiff,pdf,text).

Reply:
...also confirmed at 150dpi.

Reply:
It seems that they fixed it http://git.ghostscript.com/?p=ghostpdl.git;h=2001f469b539a5ccce301ace5d6384b6b7e46888 it wasn't included till Ghostscript 9.12 https://www.ghostscript.com/pipermail/gs-cvs/2014-March/017368.html the highest version in 14.04 is 9.10.